themes/agnosterlike/agnosterlike.theme.bash

```diff
@@ -0,0 +1,159 @@
+
+IN_VIM_PROMPT_COLOR=35
+IN_VIM_PROMPT_TEXT="vim"
+
+
+function set_rgb_color {
+  if [[ "${1}" != "-" ]]; then
+    fg="38;5;${1}"
+  fi
+  if [[ "${2}" != "-" ]]; then
+    bg="48;5;${2}"
+    [[ -n "${fg}" ]] && bg=";${bg}"
+  fi
+  echo -e "\[\033[${fg}${bg}m\]"
+}
+
+function powerline_shell_prompt {
+  SHELL_PROMPT_COLOR=${SHELL_THEME_PROMPT_COLOR};
+  SHELL_PROMPT_SYMBOLS="";
+  if [[ "$USER" == "root" ]]; then
+    SHELL_PROMPT_COLOR_FORE=${SHELL_THEME_PROMPT_COLOR_SUDO};
+    SHELL_PROMPT_SYMBOLS="${SHELL_PROMPT_SYMBOLS}⚡ "; #  \u26a1
+  fi;
+  if [[ -n "${SSH_CLIENT}" ]]; then
+    SHELL_PROMPT_SYMBOLS="${SHELL_PROMPT_SYMBOLS}${SHELL_SSH_CHAR} ";
+  fi;
+  if [[ $(jobs -l | wc -l) -gt 0 ]]; then
+    SHELL_PROMPT_SYMBOLS="${SHELL_PROMPT_SYMBOLS}$(set_rgb_color ${SHELL_GEAR_CHAR_FORE} ${SHELL_PROMPT_COLOR})${SHELL_GEAR_CHAR} ";
+  fi;
+    SHELL_PROMPT="\u@\h";
+  SHELL_PROMPT="$(set_rgb_color ${SHELL_PROMPT_COLOR_FORE} ${SHELL_PROMPT_COLOR}) ${SHELL_PROMPT_SYMBOLS}$(set_rgb_color ${SHELL_PROMPT_COLOR_FORE} ${SHELL_PROMPT_COLOR})${SHELL_PROMPT} ${normal}";
+  LAST_THEME_COLOR=${SHELL_PROMPT_COLOR};
+}
+
+function powerline_virtualenv_prompt {
+  local environ=""
+
+  if [[ -n "$CONDA_DEFAULT_ENV" ]]; then
+    environ="conda: $CONDA_DEFAULT_ENV"
+  elif [[ -n "$VIRTUAL_ENV" ]]; then
+    environ=$(basename "$VIRTUAL_ENV")
+  fi
+
+  if [[ -n "$environ" ]]; then
+    VIRTUALENV_PROMPT="$(set_rgb_color ${LAST_THEME_COLOR} ${VIRTUALENV_THEME_PROMPT_COLOR})${THEME_PROMPT_SEPARATOR}${normal}$(set_rgb_color - ${VIRTUALENV_THEME_PROMPT_COLOR}) ${VIRTUALENV_CHAR}$environ ${normal}"
+    LAST_THEME_COLOR=${VIRTUALENV_THEME_PROMPT_COLOR}
+  else
+    VIRTUALENV_PROMPT=""
+  fi
+}
+
+function powerline_scm_prompt {
+  scm_prompt_vars;
+  if [[ "${SCM_NONE_CHAR}" != "${SCM_CHAR}" ]]; then
+    if [[ "${SCM_DIRTY}" -eq 3 ]]; then
+      SCM_PROMPT="$(set_rgb_color ${SCM_THEME_PROMPT_COLOR} ${SCM_THEME_PROMPT_STAGED_COLOR})";
+      SCM_LAST_THEME_COLOR=${SCM_THEME_PROMPT_STAGED_COLOR};
+      SCM_THEICON=" ✚";
+    else
+      if [[ "${SCM_DIRTY}" -eq 2 ]]; then
+        SCM_PROMPT="$(set_rgb_color ${SCM_THEME_PROMPT_COLOR} ${SCM_THEME_PROMPT_UNSTAGED_COLOR})";
+        SCM_LAST_THEME_COLOR=${SCM_THEME_PROMPT_UNSTAGED_COLOR};
+        SCM_THEICON=" ●";
+      else
+        if [[ "${SCM_DIRTY}" -eq 1 ]]; then
+          SCM_PROMPT="$(set_rgb_color ${SCM_THEME_PROMPT_COLOR} ${SCM_THEME_PROMPT_DIRTY_COLOR})";
+          SCM_LAST_THEME_COLOR=${SCM_THEME_PROMPT_DIRTY_COLOR};
+          SCM_THEICON="";
+        else
+          SCM_PROMPT="$(set_rgb_color ${SCM_THEME_PROMPT_COLOR} ${SCM_THEME_PROMPT_CLEAN_COLOR})";
+          SCM_LAST_THEME_COLOR=${SCM_THEME_PROMPT_CLEAN_COLOR};
+          SCM_THEICON="";
+        fi;
+      fi;
+    fi;
+    if [[ "${SCM_GIT_CHAR}" == "${SCM_CHAR}" ]]; then # IS GIT
+      if echo ${SCM_BRANCH} | grep -q "detached:"; then 
+        SCM_CHAR="➦ " ;
+        SCM_BRANCH=$(git status | head -n1 | sed -E 's/.+at //g');
```

Comment:
You can check when HEAD is detached checking the env var [SCM_GIT_DETACHED](https://github.com/Bash-it/bash-it/blob/master/themes/base.theme.bash#L195), grep is not needed.

SCM_BRANCH, when detached, show the commit id relative to tag or branch names (to improve this,  can add a "flag" like the [git official prompt](https://github.com/git/git/blob/c2c5f6b1e479f2c38e0e01345350620944e3527f/contrib/completion/git-prompt.sh#L78) does to change the behavior.), simply remove the string "detached:" from the variable to prevent call git status again.

But, if you really want the commit id, use `git rev-parse --short HEAD` instead `git status`, to reduce overhead in the prompt.